Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
I don't know if we will ever have thread-specific watchpoints. You're
saying the protocol doesn't really support it, and I don't see lldb adding
that.

The problem with enabling everywhere before resuming is that when a new
thread is created, we don't stop the other threads. We just do whatever is
required for that new thread and resume that one only.

On Tuesday, August 9, 2016, Francis Ricci <notifications@github.com> wrote:

> In Sources/Target/Common/ThreadBase.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r74102382>:
>
> > @@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
> >    action(state);
> >    return writeCPUState(state);
> >  }
> > +
> > +ErrorCode ThreadBase::beforeResume() {
> > +  if (!_process->isAlive())
> > +    return kErrorProcessNotFound;
> > +
> > +  BreakpointManager *bpm = _process->hardwareBreakpointManager();
> > +  if (bpm != nullptr) {
> > +    bpm->enable();
>
> Unless we think that at some point in the future, it could be useful to
> support thread-specific watchpoints.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2a73b10db4691d10fd51d07ec46fbded7b36f6af..b32f1128fa8ceb0010cf5f79262d062aca477162#r74102382>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADHbI6Pxxy-RPjlExrQi5OVBXeHOqZyks5qeLXygaJpZM4H9FRy>
> .
>


-- 
Stephane Sezer